src/ui/src/components/blueprints/BlueprintsAutogen.vue

```diff
@@ -59,43 +59,52 @@ onMounted(() => {
  * The generation happens with simple ids (aig1, aig2, ...).
  * Component ids are altered to preserve uniqueness across blueprints.
  */
-function alterIds(components: Component[]) {
+function alterIds(actions: AutogenAction[]) {
 	const mapping: Record<string, string> = {};
 
-	// Switch ids
-
-	components.forEach((c) => {
-		const newId = generateNewComponentId();
-		mapping[c.id] = newId;
-		c.id = newId;
-	});
+	actions.forEach((a) => {
+		if (a.type == "create") {
```

Comment:
Optional, but it might make the code more readable with intermediate function, like

```ts
function alterIds(actions: AutogenAction[]) {
  // ...
  function alterIdForCreate(action: AutogenAction & { type: 'create' }) {/* ... */}

  function alterIdForLink(action: AutogenAction & { type: 'link' }) {/* ... */}

  actions.forEach((a) => {
		if (a.type == "create") alterIdForCreate(a);
       else if (a.type == "list") alterIdForLink(a);
  });
}
```